#!/usr/bin/env python
PACKAGE = "ugv_path_tracker"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("a", double_t, 0, "rotations ", 2.5,0,10)
gen.add("b", double_t, 0, "linear", 3,0,10)
gen.add("b_back", double_t, 0, "linear back speed factor", 3,0,10)
gen.add("dist_margin", double_t, 0, "Margin to consider a goal as reached ", 15,0,90)
gen.add("angular_max_speed", double_t, 0, "Angular max speed(radians)", 1.2,0.2,1.8)
gen.add("linear_max_speed", double_t, 0, "Max speed in direction +x", 0.3,0.2,1)
gen.add("linear_max_speed_back", double_t, 0, "Max speed in direction -x", 0.3,0.2,1)
#gen.add("angle_margin", double_t, 0, "Margin to consider an orientarion as reached ", 15,0,20)
gen.add("do_navigate", bool_t, 0, "To navigate or to not navigate",True)
gen.add("holonomic", bool_t, 0, "Holonomic or not",True)


gen.add("angle1", double_t, 0, "angle1 ", 20,0,180)
gen.add("angle2", double_t, 0, "angle2", 65,0,180)
gen.add("angle3", double_t, 0, "angle3", 15,0,180)

exit(gen.generate(PACKAGE, "ugv_path_tracker", "PathTracker"))